go/adbc/driver/databricks/connection.go

```diff
@@ -0,0 +1,166 @@
+	// Default Catalog name (optional)
+	catalog string
+	// Default Schema name (optional)
+	dbSchema string
+}
+
+func sanitizeSchema(schema string) (string, error) {
+	// TODO(felipecrv): sanitize databricks schemas
+	return schema, nil
+}
+
+func (conn *connectionImpl) StatementExecution() sql.StatementExecutionInterface {
+	if conn.client == nil {
+		return nil
+	}
+	return conn.client.StatementExecution
+}
+
+// driverbase.CurrentNamespacer {{{
+
+func (conn *connectionImpl) GetCurrentCatalog() (string, error) {
+	return conn.catalog, nil
+}
+
+func (conn *connectionImpl) GetCurrentDbSchema() (string, error) {
+	return conn.dbSchema, nil
+}
+
+func (conn *connectionImpl) SetCurrentCatalog(value string) error {
+	conn.catalog = value
+	return nil
+}
+
+func (conn *connectionImpl) SetCurrentDbSchema(value string) error {
+	sanitized, err := sanitizeSchema(value)
+	if err != nil {
+		return err
+	}
+	conn.dbSchema = sanitized
+	return nil
+}
+
+// }}}
+
+// driverbase.TableTypeLister {{{
+
+func (c *connectionImpl) ListTableTypes(ctx context.Context) ([]string, error) {
+	// TODO(felipecrv): implement ListTableTypes
+	return []string{
+		"TABLE",
+		"VIEW",
+	}, nil
+}
+
+// }}}
+
+// driverbase.AutocommitSetter {{{
+
+func (conn *connectionImpl) SetAutocommit(enabled bool) error {
+	if enabled {
+		return nil
+	}
+	return adbc.Error{
+		Code: adbc.StatusNotImplemented,
+		Msg:  "SetAutocommit to `false` is not yet implemented",
+	}
+}
+
+// }}}
+
+// driverbase.DbObjectsEnumerator {{{
+
+func (conn *connectionImpl) GetCatalogs(ctx context.Context, catalogFilter *string) ([]string, error) {
+	catalogPattern, err := internal.PatternToRegexp(catalogFilter)
+	if err != nil {
+		return nil, err
+	}
+	if catalogPattern == nil {
+		catalogPattern = internal.AcceptAll
+	}
+
+	res := make([]string, 0)
+	// TODO: implement GetCatalogs
+	// for _, catalog := range conn.client.Catalogs() {
+	//   if catalogPattern.MatchString(catalog) {
+	//     res = append(res, catalog)
+	//   }
+	// }
+	return res, nil
+}
+
+func (conn *connectionImpl) GetDBSchemasForCatalog(ctx context.Context, catalog string, schemaFilter *string) ([]string, error) {
+	schemaPattern, err := internal.PatternToRegexp(schemaFilter)
+	if err != nil {
+		return nil, err
+	}
+	if schemaPattern == nil {
+		schemaPattern = internal.AcceptAll
+	}
+
+	res := make([]string, 0)
+	// TODO: implement GetDBSchemasForCatalog
+	return res, nil
+}
+
+func (conn *connectionImpl) GetTablesForDBSchema(ctx context.Context, catalog string, schema string, tableFilter *string, columnFilter *string, includeColumns bool) ([]driverbase.TableInfo, error) {
+	tablePattern, err := internal.PatternToRegexp(tableFilter)
+	if err != nil {
+		return nil, err
+	}
+	if tablePattern == nil {
+		tablePattern = internal.AcceptAll
+	}
+
+	res := make([]driverbase.TableInfo, 0)
+	// TODO: implement GetTablesForDBSchema
+	return res, nil
+}
+
+// }}}
+
+// NewStatement initializes a new statement object tied to this connection
```

Comment:
nit: some (public) functions have comments and some do not. Did not see any pattern.

Reply:
I want to call out that the statement is tied to the connection. Other functions are implementations of a documented interface.

Reply:
ok was not clear from the context. but fine